crabs/detection_tracking/detection_utils.py

```diff
@@ -226,65 +229,243 @@ def calculate_iou(box1, box2) -> float:
-    iou_threshold : float
-        The intersection over union threshold for considering a match.
-    frame_copy : np.ndarray
-        A copy of the input frame for drawing bounding boxes.
+    Parameters:
+    -----------
+    bbox : np.ndarray
+        A numpy array containing the bounding box coordinates
+        (xmin, ymin, xmax, ymax, id).
+    frame : np.ndarray
+        The frame to which the bounding box belongs.
+    frame_name : str
+        The name of the frame.
+    csv_writer : Any
+        The CSV writer object to write the annotation.
     """
 
-    for object_id, gt_data in ground_truths.items():
-        gt_annotations = gt_data.get("annotations", [])
-        for gt_box in gt_annotations:
-            if gt_box["image_id"] == frame_number:
-                x_gt, y_gt, width_gt, height_gt = gt_box["bbox"]
-                x2_gt, y2_gt = x_gt + width_gt, y_gt + height_gt
-                x_gt, y_gt, x2_gt, y2_gt = (
-                    int(x_gt),
-                    int(y_gt),
-                    int(x2_gt),
-                    int(y2_gt),
-                )
+    # Bounding box geometry
+    xmin, ymin, xmax, ymax, id = bbox
+    width_box = int(xmax - xmin)
+    height_box = int(ymax - ymin)
+
+    # Add to csv
+    csv_writer.writerow(
+        (
+            frame_name,
+            frame.size,
+            '{{"clip":{}}}'.format("123"),
+            1,
+            0,
+            '{{"name":"rect","x":{},"y":{},"width":{},"height":{}}}'.format(
+                xmin, ymin, width_box, height_box
+            ),
+            '{{"track":"{}"}}'.format(id),
+        )
+    )
 
-                for bbox in tracked_boxes:
-                    x1_track, y1_track, x2_track, y2_track, _ = bbox
-                    iou = calculate_iou(
-                        [x_gt, y_gt, x2_gt, y2_gt],
-                        [x1_track, y1_track, x2_track, y2_track],
-                    )
 
-                    if iou > iou_threshold:
-                        cv2.rectangle(
-                            frame_copy,
-                            (x_gt, y_gt),
-                            (x2_gt, y2_gt),
-                            (0, 255, 0),
-                            2,
-                        )
-                        id_label = f"id : {object_id}"
-                        draw_bbox(
-                            frame_copy,
-                            int(x1_track),
-                            int(y1_track),
-                            int(x2_track),
-                            int(y2_track),
-                            (0, 0, 255),
-                            id_label,
-                        )
+def save_frame_and_csv(
+    video_file_root: str,
+    tracking_output_dir: Path,
+    tracked_boxes: List[List[float]],
+    frame: np.ndarray,
+    frame_number: int,
+    csv_writer: Any,
+    save_plot: bool = True,
+) -> np.ndarray:
+    """
+    Common functionality for saving frames and CSV
```

Comment:
suggestion for a bit more context:
```suggestion
    Save tracked bounding boxes data as frames and CSV file
```
maybe we can add: this can be useful for manually correcting annotations